onelogin/get.go

```diff
@@ -107,30 +109,80 @@ func Get(app, provider string) (*awsprovider.Credentials, error) {
 		fmt.Scanln(&selection)
 
 		deviceID = fmt.Sprintf("%v", devices[selection-1].DeviceId)
+		deviceType = devices[selection-1].DeviceType
+
 	} else {
 		deviceID = fmt.Sprintf("%v", devices[0].DeviceId)
+		deviceType = devices[0].DeviceType
 	}
 
-	fmt.Print("Please enter the OTP from your MFA device: ")
-	var otp string
-	fmt.Scanln(&otp)
+	var samlAssertion string
+
+	var pushSupported = false
+
+	if deviceType == "OneLogin Protect" {
+		// push is supported by the selected OTP device
+		// if other are supporting push they should be added here.
+		pushSupported = true
+		s.Start()
+		pMfa := VerifyFactorParams{
+			AppId:       a.ID,
+			DeviceId:    deviceID,
+			StateToken:  st,
+			OtpToken:    "",
+			DoNotNotify: false,
+		}
 
-	// Verify MFA
-	pMfa := VerifyFactorParams{
-		AppId:      a.ID,
-		DeviceId:   deviceID,
-		StateToken: st,
-		OtpToken:   otp,
-	}
+		rMfa, err := c.VerifyFactor(token, &pMfa)
+		if err != nil {
+			return nil, err
+		}
+		pMfa.DoNotNotify = true
+		timeout := 30
+		interval := 4
+		fmt.Printf(" %v\r", rMfa.Status.Message)
+		for rMfa.Status.Type == "pending" && timeout > 0 {
```

Comment:
I would have liked to fail fast here in case the user presses "reject" on the app, however looks like the status remains "pending" both when the user doesn't do anything and when they press "reject"...
Any idea if we can get a "reject" indication?

Reply:
nope. if the status changes we're leaving the loop. Maybe a future API update. ;-)

Reply:
The thing is - pressing "reject" doesn't make us leave the loop, we only leave when it times out...